### PR TITLE
Update the test app to be able to save ADAL logs

### PR DESCRIPTION
**Pull request by heidijinxujia**
Issue#738
- add the feature in the sample app to collect ADAL logs
- add the button in the UI to show the ADAL logs
- modify button for "invalidateRT"
- add button for "invalidateFRT"
- add button for "readLogs"
- fix the bug for invalidating the FRT

### CODE REVIEW

**Comment by heidijinxujia** [comment]
Hi @weijjia @kpanwar I've addressed the comments. Please let me know if there is anything more you would like to add into the test app. Thanks a lot. :)

File: automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/MainActivity.java [hunk 1]
```diff
@@ -50,17 +53,28 @@
...
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-
         mContext = getApplicationContext();
+        
+        mADALLogs = new StringBuffer();
+        Logger.getInstance().setExternalLogger(new Logger.ILogger() {
```
Comment (weijjia) [inline automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/MainActivity.java]: I think you can set the external logger in the Application class, that way we won't need to have it in both MainActivity and SigninActivity.

File: automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/ResultActivity.java [hunk 2]
```diff
@@ -61,6 +66,10 @@ protected void onCreate(Bundle savedInstanceState) {
         }
         mTextView.setText(resultText);
 
+        mLogView = (TextView) findViewById(R.id.adalLogs);
+        mLogView.setMovementMethod(new ScrollingMovementMethod());
```
Comment (weijjia) [inline automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/ResultActivity.java]: do you need the log text box to be scrollable?

File: automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/ResultActivity.java [hunk 3]
```diff
@@ -45,6 +48,8 @@
 public class ResultActivity extends AppCompatActivity {
 
     private TextView mTextView;
+    private TextView mLogView;
+    private StringBuffer mADALLogs = new StringBuffer();
```
Comment (weijjia) [inline automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/ResultActivity.java]: why the mADALLogs cannot be declared locally?

File: automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/ResultActivity.java (hunk 2 above)
Comment (heidijinxujia) [inline automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/ResultActivity.java]: It depends on the requirement. We don't need to set the movement method for the text box if we don't need to show all the logs.

File: automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/MainActivity.java (hunk 1 above)
Comment (heidijinxujia) [inline automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/MainActivity.java]: Got it, thanks @weijjia :)

File: automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/ResultActivity.java (hunk 2 above)
Comment (weijjia) [inline automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/ResultActivity.java]: My understanding is that the textbox is used to save the logs, not for us to actually show it. IMHO, there is no need to set the movement method. :)

File: automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/ResultActivity.java (hunk 3 above)
Comment (heidijinxujia) [inline automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/ResultActivity.java]: Addressed. Moved to the application class.

File: automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/ResultActivity.java (hunk 2 above)
Comment (heidijinxujia) [inline automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/ResultActivity.java]: Addressed. Thank you, @weijjia

**Comment by kpanwar** [comment]
:shipit:

**Pull request merged**